packages/eslint-plugin/tests/rules/no-unnecessary-type-constraint.test.ts

```diff
@@ -88,10 +108,85 @@ function data<T extends TODO>() {}
           endColumn: 28,
           column: 15,
           line: 1,
+          suggestions: [
+            {
+              messageId: 'removeUnnecessaryConstraint',
+              output: noFormat`const data = <T,>() => {};`,
+            },
+          ],
+        },
+      ],
+      filename: 'react.tsx',
+    },
+    {
+      code: noFormat`const data = <T extends any,>() => {};`,
```

Comment:
Checking for commas tends to make me nervous because it's easy to write logic that's sensitive to the amount of spaces around tokens such as the `,`. Although the rule works as-is, it'd be easy to slip up in the future. 

Could you add a few test cases for spacing around the comma? Maybe:

* `const data = <T extends any ,>() => {};`,
* `const data = <T extends any, >() => {};`,
* `const data = <T extends any  ,  >() => {};`,